### ISSUE TITLE
How to indicate that a schema using a transform or preprocess can be nullable / optional?

### ISSUE THREAD

**Issue by codeinabox** [asteasolutions/zod-to-openapi#143]
Is there a way to create a nullable transform with schema version 3.0.0? I had the following schema, which used coerce:

```ts
const aSchema = z.object({
  id: z.string(),
  date: z.coerce
    .string()
    .nullable()
    .openapi({ example: '2023-04-12', description: 'YYYY-MM-DD' }),
});
```

This correctly set `nullable: true` on the generated document. However due a bug with coerce https://github.com/colinhacks/zod/issues/1760, I've had to switch to the following transform schema and I am having trouble getting it to say the field can be null in the response

```ts
const aSchema = z.object({
  id: z.string(),
  date: z
    .any()
    .transform((val: LocalDate | null) => (val ? String(val) : null))
    .nullable()
    .openapi({ example: '2023-04-12', description: 'YYYY-MM-DD', type: 'string' }),
});
```

**Comment by codeinabox** [comment]
I've also tried this as a preprocess and have the same issue:

```ts
const aSchema = z.object({
  id: z.string(),
  date: z
    .preprocess(val => (val ? String(val) : null), z.string())
    .nullable()
    .openapi({ example: '2023-04-12', description: 'YYYY-MM-DD' }),
});
```

**Comment by AGalabov** [comment]
@codeinabox  thank you for using the library and helping us identify issues to make it better. I'll answer on issue in two ways:

1. We've taken the decision that when a `type` property is passed then all of the documentation is the responsibility of the user - i.e leaving it up to him since he is modifying the main field. That is why the nullable is not being generated - you can add it manually.

2. However for the coerce type this is no longer the issue. There is a problem on our side in this case. I'll open up a PR soon that should fix it.

**Comment by AGalabov** [comment]
@codeinabox some changes have been released with [v5.1.0](https://github.com/asteasolutions/zod-to-openapi/releases/tag/v5.1.0). With them the `preprocess` example should work.
